File: 231213/원 모양으로 되어있는 방/a-room-in-a-circle.py
N = int(input())
personList = [int(input()) for _ in range(N)]

import sys

minDis = sys.maxsize

for start in range(len(personList)):
    tempSum = 0

    for i in range(N):
        currentIdx = (start + i) % N

        tempSum += personList[currentIdx] * i
    
    minDis = min(minDis, tempSum)

print(minDis)

File: 231213/원 모양으로 되어있는 방/a-room-in-a-circle.js
const fs = require('fs');
const inputAll = () => fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const print = (...args) => console.log(...args);

let [N, ...arr] = inputAll();

N = Number(N);
arr = arr.map(elm => Number(elm));

let minSum = Infinity;

for(let start = 0; start < arr.length; start++){

    let tempSum = 0;

    for(let dis = 0; dis < arr.length; dis++){
        const currentIdx = (start + dis) % N;

        tempSum += arr[currentIdx] * dis;
    }

    minSum = Math.min(minSum, tempSum);
}

print(minSum);